src/BlocktopmenuNode.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+class BlocktopmenuNode implements ArrayAccess, Countable, IteratorAggregate
+{
+    const TYPE_CATEGORY = 'category';
+    const TYPE_CATEGORY_THUMBNAILS = 'category-thumbnails';
+    const TYPE_PRODUCT = 'product';
+    const TYPE_MANUFACTURERS = 'manufacturers';
+    const TYPE_MANUFACTURER = 'manufacturer';
+    const TYPE_SUPPLIERS = 'suppliers';
+    const TYPE_SUPPLIER = 'supplier';
+    const TYPE_CMS = 'cms';
+    const TYPE_CMS_CATEGORY = 'cms-category';
+    const TYPE_SHOP = 'shop';
+    const TYPE_LINK = 'link';
+
+    private $type;
+    private $selected = false;
+    private $data = array();
+    private $children = array();
+
+    public function __construct($type, array $data = array())
+    {
+        $this->type = $type;
+        $this->data = $data;
+    }
+
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    public function getData($name)
+    {
+        return isset($this->data[$name]) ? $this->data[$name] : null;
+    }
+
+    public function addChild(BlocktopmenuNode $node)
+    {
+        $this->children[] = $node;
+
+        return $this;
+    }
+
+    public function setSelected($selected = true)
+    {
+        $this->selected = $selected;
+
+        return $this;
+    }
+
+    public function isSelected()
+    {
+        return $this->selected;
+    }
+
+    public function getChildren()
+    {
+        return $this->children;
+    }
+
+    public function offsetExists($offset)
+    {
+        return null !== $this->offsetGet($offset);
+    }
+
+    public function offsetGet($offset)
+    {
+        switch ($offset) {
+            case 'selected' :
+                return $this->selected;
+            case 'children' :
+                return $this->children;
+            case 'data' :
+                return $this->data;
+                break;
+            case 'type' :
+                return $this->type;
+                break;
```

Comment:
It's nice that the model is flexible enough so that we can store an ID for the entities for which it makes sense.
(eg, the link ID, or the new_window attribute).
But, is the foreach loop in blocktopmenu.tpl active/relevant?
ie: I would want:
```<a href="{$item.data.link}" title="{$item.data.name}" id="blocktopmenu_menu_{$item.data.type}_{$item.data.id}>{$item.data.name}</a>```
Would I change it in the markup hardcoded inside SuperfishNodeVisitor.php? Would I create my own class? Or would I have to change/override the template?